### PR TITLE
Refactor httpd ‒ split to content & http parts

### PR DESCRIPTION
**Pull request by vorner**
This splits the httpd related handling into the http parts and content related parts.

The primary intention is to unblock unit & fuzz tests on the http part (which is exposed to the outside without authentication). It also draws the boundary between parts in somewhat thicker line and will make our life a little bit easier once we start replacing the HTTP server itself… but the boundary is probably in a suboptimal place and will eventually move ‒ this is what the code was doing now and I didn't want to introduce big or functional changes. In other words, this is supposed to be gradual improvement but is far from the ideal.

### CODE REVIEW

File: lib/WUI/http/handler_defaults.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#define GET_WRAPPER(NAME)                                                                       \
+    static void handler_##NAME(struct HttpHandlers *unused, char *buffer, size_t buffer_size) { \
+        (void)unused;                                                                           \
+        NAME(buffer, buffer_size);                                                              \
+    }
+GET_WRAPPER(get_printer);
+GET_WRAPPER(get_version);
+GET_WRAPPER(get_job);
+GET_WRAPPER(get_files);
+#undef GET_WRAPPER
```
Comment (dragomirecky) [inline lib/WUI/http/handler_defaults.c:16]: /question/ As I understand it, the `lib/WUI/http` directory contains implementation of a generic http client/server (which shouldn't have any knowledge of our specific endpoints). In such case, wouldn't it be cleaner to move this file under `lib/WUI` and name it – for example – `wui_handlers.c`?

File: lib/WUI/http/handler_defaults.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        .uri = "/api/job",
+        .handler = handler_get_job,
+    },
+    {
+        .uri = "/api/files",
+        .handler = handler_get_files,
+        .prefix = true, // Any uri starting with /api/files is fine (for now).
+    },
+    {}, // Sentinel
+};
```
Comment (dragomirecky) [inline lib/WUI/http/handler_defaults.c:49]: Just curious – do we use the `anonymous` flag somewhere? Is it the intention that it exists and isn't used anywhere (from what I've seen)

**Review by dragomirecky — DISMISSED** [review]

File: lib/WUI/http/handler_defaults.c (hunk 1 above)
Comment (vorner) [reply lib/WUI/http/handler_defaults.c]: Well, the file structure around WUI is its own chapter (some parts are in `src`, some in `lib`…) and it's not even fully true that the http server would be fully agnostic of the content (the POST part ☹). But I guess this is as good time as any to help it a bit.

File: lib/WUI/http/handler_defaults.c (hunk 2 above)
Comment (vorner) [reply lib/WUI/http/handler_defaults.c]: Currently, no, not yet. I expect to use it in tests and mocks. We also have some files as static files and these are anonymous, but that doesn't go through this mechanism yet.

**Review by dragomirecky — APPROVED** [review]

**Pull request merged**